### ISSUE TITLE
How to resend message code?

### ISSUE THREAD

**Issue by lslima91** [invertase/react-native-firebase#721]
Currently ForceResendingToken doesn't seem to be exposed to Js.

**Comment by chrisbianca** [comment]
@lslima91 You're right, we're missing this - I've marked it as a feature request which we'll look to add in the future.

**Comment by perrosnk** [comment]
Any updates on this?

**Comment by bhakta0007** [comment]
+1

this is an important feature. Can we have this prioritized please? Even a patch will do, while we wait for this to be available in a release build.

**Comment by adirzoari** [comment]
wait for this feature also. any update?

**Comment by DrFlavour** [comment]
+1 This would be really useful.

**Comment by Salakar** [comment]
@lslima91 @perrosnk @bhakta0007 @adirzoari @DrFlavour Have just pushed up Android support for this to master, would be great if you could try it out and let me know all is ok.

```typescript
signInWithPhoneNumber(phoneNumber: string, forceResend?: boolean): Promise<ConfirmationResult>;


verifyPhoneNumber(
  phoneNumber: string,
  autoVerifyTimeoutOrForceResend?: number | boolean,
  forceResend?: boolean,
): PhoneAuthListener;
```

Worth noting that forceResend will do nothing on the first attempt and signing in with/verifying a phone number - subsequent requests will use it however.

---
Loving `react-native-firebase` and the support we provide? Please consider supporting us with any of the below:

  - 👉  Back financially via [Open Collective](https://opencollective.com/react-native-firebase/donate)  
  - 👉  Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter 
  - 👉  Star this repo on GitHub ⭐️

**Comment by adirzoari** [comment]
@Salakar  I would be happy to check it.
I just need to call verifyPhoneNumber with same values but in forceResend =true?

**Comment by Salakar** [comment]
@adirzoari yes, something like:

```js
verifyPhoneNumber('12345679687', true);
// or with auto verify timeout of 60 secs:
verifyPhoneNumber('12345679687', 60, true);
```

---
Loving `react-native-firebase` and the support we provide? Please consider supporting us with any of the below:

  - 👉  Back financially via [Open Collective](https://opencollective.com/react-native-firebase/donate)  
  - 👉  Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter 
  - 👉  Star this repo on GitHub ⭐️

**Comment by adirzoari** [comment]
auto verify timeout will send automatically sms in 60 seconds?

**Comment by Salakar** [comment]
Auto verify timeout is how long on android the automatic sms verification should wait for a message to be received and auto verify itself - as in without the user having to input the code, before failing and falling back to waiting for the user to submit the code manually.

If you don't need to change this you can leave it 😛 

---
Loving `react-native-firebase` and the support we provide? Please consider supporting us with any of the below:

  - 👉  Back financially via [Open Collective](https://opencollective.com/react-native-firebase/donate)  
  - 👉  Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter 
  - 👉  Star this repo on GitHub ⭐️

**Comment by Salakar** [comment]
Support for this is now live in the [v4.3.0 release](https://github.com/invertase/react-native-firebase/releases/tag/v4.3.0).

---
Loving `react-native-firebase` and the support we provide? Please consider supporting us with any of the below:

  - 👉  Back financially via [Open Collective](https://opencollective.com/react-native-firebase/donate)  
  - 👉  Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter 
  - 👉  Star this repo on GitHub ⭐️

**Comment by SercanShortcut** [comment]
Here is a useful blog post for javascript developers:

"How to resend SMS verification in Firebase Phone Authentication?"
https://storksnestblog.wordpress.com/2019/06/16/how-to-resend-sms-verification-in-firebase-phone-authentication/

**Comment by anastely** [comment]
Hello @Salakar, what's deferent between `verifyPhoneNumber()` and `signInWithPhoneNumber()` in all of these i will recived an SMS verification code ?
